Zend/zend_execute.c

```diff
@@ -853,7 +853,7 @@ static zend_bool zend_verify_weak_scalar_type_hint(zend_uchar type_hint, zval *a
 				return 0;
 			}
 			zval_ptr_dtor(arg);
-			ZVAL_BOOL(arg, dest);
+			ZVAL_BOOL(ret, dest);
```

Comment:
I can't remember the exact circumstances of that logic, but it has to do with the different types of zval (var, tmp, cv, const) we can get in zend_std_write_property() and zend_assign_to_variable() and some of those must not be modified.

Possibly this can be simplified with a preliminary ZVAL_COPY_VALUE() at these locations. I need to check that.